vowpalwabbit/sample_pdf.cc

```diff
@@ -112,13 +112,16 @@ LEARNER::base_learner* sample_pdf_setup(options_i& options, vw& all)
   if (!options.add_parse_and_check_necessary(new_options)) return nullptr;
 
   LEARNER::base_learner* p_base = setup_base(options, all);
-  auto p_reduction = scoped_calloc_or_throw<sample_pdf>();
+  auto p_reduction = VW::make_unique<sample_pdf>();
   p_reduction->init(as_singleline(p_base), all.get_random_state());
 
-  LEARNER::learner<sample_pdf, example>& l = init_learner(p_reduction, as_singleline(p_base), predict_or_learn<true>,
-      predict_or_learn<false>, 1, prediction_type_t::action_pdf_value, all.get_setupfn_name(sample_pdf_setup));
+  auto* l = make_reduction_learner(std::move(p_reduction), as_singleline(p_base), predict_or_learn<true>,
+      predict_or_learn<false>, all.get_setupfn_name(sample_pdf_setup))
+                .set_prediction_type(prediction_type_t::action_pdf_value)
+                .set_label_type(label_type_t::simple)
```

Comment:
This learner does not care what the label type is, and should therefore assume whatever bases label type is. It would be better if we could mark it as explicitly pass through but not sure the best way to do that.

For now the best way to communicate that base's label type should be used is actually to omit `set_label_type`. Can you add a comment in place of this line saying it assumes base's label type?

Reply:
Just added in a comment